torch_np/_detail/implementations.py

```diff
@@ -19,11 +21,125 @@ def tensor_equal(a1_t, a2_t, equal_nan=False):
     return bool(result.all())
 
 
+def tensor_equiv(a1_t, a2_t):
+    # *almost* the same as tensor_equal: _equiv tries to broadcast, _equal does not
+    try:
+        a1_t, a2_t = torch.broadcast_tensors(a1_t, a2_t)
+    except RuntimeError:
+        # failed to broadcast => not equivalent
+        return False
+    return tensor_equal(a1_t, a2_t)
+
+
+def tensor_isclose(a, b, rtol=1.0e-5, atol=1.0e-8, equal_nan=False):
+    dtype = _dtypes_impl.result_type_impl((a.dtype, b.dtype))
+    a = a.to(dtype)
+    b = b.to(dtype)
+    result = torch.isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+    return result
+
+
+# ### is arg real or complex valued ###
+
+
+def tensor_iscomplex(x):
+    if torch.is_complex(x):
+        return torch.as_tensor(x).imag != 0
+    result = torch.zeros_like(x, dtype=torch.bool)
+    if result.ndim == 0:
+        result = result.item()
+    return result
+
+
+def tensor_isreal(x):
+    if torch.is_complex(x):
+        return torch.as_tensor(x).imag == 0
+    result = torch.zeros_like(x, dtype=torch.bool)
+    if result.ndim == 0:
+        result = result.item()
+    return result
+
+
+def tensor_real_if_close(x, tol=100):
+    if not torch.is_complex(x):
+        return x
+    mask = torch.abs(x.imag) < tol * torch.finfo(x.dtype).eps
+    if mask.all():
+        return x.real
+    else:
+        return x
+
+
+# ### math functions ###
+
+
+def tensor_angle(z, deg=False):
+    result = torch.angle(z)
+    if deg:
+        result *= 180 / torch.pi
```

Comment:
```suggestion
        result = result * (180 / torch.pi)
```
Let's try to avoid in-place operations whenever possible, as they are not differentiable in PyTorch.